src/goTest.ts

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+import cp = require('child_process');
+import path = require('path');
+import vscode = require('vscode');
+import util = require('util');
+import { getGoRuntimePath } from './goPath'
+
+/**
+* Executes the unit test at the primary cursor using `go test`. Output
+* is sent to the 'Go' channel.
+* 
+* TODO: go test returns filenames with no path information for failures,
+* so output doesn't produce navigable line references.
+*/
+export function testAtCursor() {
+	var editor = vscode.window.activeTextEditor;
+	// TODO: This command should be returning a SymbolInformation[] but is instead
+	// returning an object with slightly different fields which must be adapted.
+	vscode.commands.executeCommand<any[]>('vscode.executeDocumentSymbolProvider', editor.document.uri).then(res => {
+		var testFunction: string;
+		// Find any test function containing the cursor.
+		res.forEach(obj => {
```

Comment:
That return won't break early since it just returns from the callback for that particular item in the collection.  There is no way to break early from a `.forEach` in JavaScript unfortunately.  I believe you can use `for..of` in TS instead here. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of